app/models/grade_entry_item.rb

```diff
@@ -88,13 +88,17 @@ def self.create_or_update_from_csv_rows(names, totals, grade_entry_form, overwri
     names.shift
     totals.shift
 
+    grade_entrys = []
+
     (names.size).times do |i|
-      grade_entry_item = grade_entry_form.grade_entry_items.find_or_create_by(name: names[i])
+      grade_entry_item = grade_entry_form.grade_entry_items.find_by(name: names[i])
```

Comment:
This is okay, but you can use [first_or_initialize](http://www.mitchcrowe.com/10-most-underused-activerecord-relation-methods/) instead.